grammars/rescript.tmLanguage.json

```diff
@@ -119,84 +119,87 @@
-          "endCaptures": {
-            "1": {
-              "name": "punctuation.definition.string.end"
-            }
-          },
-          "patterns": [
-            {
-              "include": "#string-character-escape"
-            }
-          ]
+          "include": "#string-character-escape"
+        }
+      ]
+    },
+    "template": {
+      "name": "string.template",
+      "begin": "([a-z_][0-9a-zA-Z_]*)?(`)",
+      "end": "(?<!\\\\)`",
+      "beginCaptures": {
+        "1": {
+          "name": "entity.name.function"
         },
-        {
-          "name": "string.template",
-          "begin": "([a-z_][0-9a-zA-Z_]*)?(`)",
-          "end": "(?<!\\\\)`",
-          "beginCaptures": {
-            "1": {
-              "name": "entity.name.function"
-            },
-            "2": {
-              "name": "punctuation.definition.string.template.begin"
-            }
-          },
-          "endCaptures": {
-            "1": {
-              "name": "punctuation.definition.string.template.end"
-            }
-          },
-          "patterns": [
-            {
-              "include": "#string-character-escape"
-            },
-            {
-              "name": "meta.template.expression",
-              "begin": "\\$\\{",
-              "beginCaptures": {
-                "0": {
-                  "name": "punctuation.definition.template-expression.begin"
-                }
-              },
-              "end": "\\}",
-              "endCaptures": {
-                "0": {
-                  "name": "punctuation.definition.template-expression.end"
-                }
-              },
-              "patterns": [
-                {
-                  "match": "[a-z_][0-9a-zA-Z_]*"
-                },
-                {
-                  "include": "#operator"
-                },
-                {
-                  "include": "#punctuations"
-                },
-                {
-                  "include": "#string"
-                }
-              ]
-            }
-          ]
+        "2": {
+          "name": "punctuation.definition.string.template.begin"
         }
+      },
+      "endCaptures": {
+        "1": {
+          "name": "punctuation.definition.string.template.end"
+        }
+      },
+      "patterns": [
+        { "include": "#string-character-escape" },
+        { "include": "#template-substitution-element" }
+      ]
+    },
+    "template-substitution-element": {
+      "name": "meta.template.expression",
+      "begin": "\\$\\{",
+      "beginCaptures": {
+        "0": {
+          "name": "punctuation.definition.template-expression.begin"
+        }
+      },
+      "end": "\\}",
+      "endCaptures": {
+        "0": {
+          "name": "punctuation.definition.template-expression.end"
+        }
+      },
+      "patterns": [
+        { "include": "#commentLine" },
+        { "include": "#commentBlock" },
+        { "include": "#operator" },
+        { "include": "#punctuations" },
```

Comment:
Instead of highlighting operators and punctuations separately, they should be part of `expression`